### PR TITLE
Task/04 Styling

### PR DESCRIPTION
**Pull request by roman-orlovskyi-moc**
Acceptance Criteria for 1
- [x] Mentor is added to pr
- [x] AC copied to pr description
- [x] Link to deployed version is added to pr
- [x] Added layout of `Search Bar`
- [ ] Rewritten all styles with `Module Styles`
- [x] Implemented `dynamic theme change` logic

Acceptance Criteria for 1*
- [ ] Implemented logic for fetching `user's theme`.
- [x] Saved theme to the `Local storage`.
Note: if we have saved theme in `Local storage`, we use it first

[deployed version](https://roman-orlovskyi-moc.github.io/fe-react-2024/)

### CODE REVIEW

File: src/components/products-filter-bar/products-filter-bar.module.css [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  align-items: center;
+  grid-template-columns: 275px 1fr 260px;
+  grid-template-areas: "search-form category-filter sort-wrapper";
+  margin-bottom: 50px;
+  gap: 20px;
+  font-size: 14px;
+  font-weight: 600;
+}
+
+.productsFilterSortWrapper {
```
Comment (vladyslav-yermolin-moc) [inline src/components/products-filter-bar/products-filter-bar.module.css:12]: Hey, @roman-orlovskyi-moc, your layout in the middle values of the width looks a bit odd.
I propose to decrease gap between header and filters, align filters to upper edge, and align chips (Clothes, Shoes,...) to the left side
image

File: src/App.module.css [hunk 2]
```diff
@@ -0,0 +1,74 @@
+:global :root.dark {
```
Comment (vladyslav-yermolin-moc) [inline src/App.module.css]: these common styles could live in the [src/index.css](https://github.com/roman-orlovskyi-moc/fe-react-2024/pull/4/files#diff-85da366246340e911d7a0eb9153e64137a1b31f0686c74d9aef9ae4f032cb09e)

File: src/components/categories-filter/CategoriesFilter.component.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+import styles from './categories-filter.module.css';
+
+interface CategoriesFilterProps {
+    categoryIds?: number[];
+    onCategoriesChange: (categoryIds: number[]) => void;
+}
+
+export const CategoriesFilterComponent: React.FC<CategoriesFilterProps> = ({ categoryIds, onCategoriesChange }) => {
+    const [checkedCategories, setCheckedCategories] = useState<CheckedCategoriesProps>(() => transformCheckedCategories(categoryIds || []));
```
Comment (vladyslav-yermolin-moc) [inline src/components/categories-filter/CategoriesFilter.component.tsx]: `Props` we add to React component props, here it confused

```suggestion
    const [checkedCategories, setCheckedCategories] = useState<CheckedCategories>(() => transformCheckedCategories(categoryIds || []));
```

File: src/components/categories-filter/CategoriesFilter.component.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    };
+
+    return (
+        <div className={styles.categoryFilterWrapper}>
+            <CategoriesDataProviderComponent>
+                {({ categories }) => (
+                    <>
+                        {categories.map((categoryData) => (
+                            <div key={categoryData.id}>
+                                <input
```
Comment (vladyslav-yermolin-moc) [inline src/components/categories-filter/CategoriesFilter.component.tsx]: Would be good to add a bit of semantic
```suggestion
                    <ul>
                        {categories.map((categoryData) => (
                            <li key={categoryData.id}>
                                <input
```

**Review by vladyslav-yermolin-moc — COMMENTED** [review]
Cool work👏
You almost invented your own router. Useful experiense😎

File: src/components/products-filter-bar/products-filter-bar.module.css (hunk 1 above)
Comment (roman-orlovskyi-moc) [reply src/components/products-filter-bar/products-filter-bar.module.css:12]: fixed

File: src/App.module.css (hunk 2 above)
Comment (roman-orlovskyi-moc) [reply src/App.module.css]: fixed

File: src/components/categories-filter/CategoriesFilter.component.tsx (hunk 3 above)
Comment (roman-orlovskyi-moc) [reply src/components/categories-filter/CategoriesFilter.component.tsx]: fixed

File: src/components/categories-filter/CategoriesFilter.component.tsx (hunk 4 above)
Comment (roman-orlovskyi-moc) [reply src/components/categories-filter/CategoriesFilter.component.tsx]: fixed

**Review by vladyslav-yermolin-moc — APPROVED** [review]

**Pull request merged**